src/Storage/Storage.Management/Models/PSBlobInventoryPolicy.cs

```diff
@@ -55,32 +55,29 @@ public PSBlobInventoryPolicy(BlobInventoryPolicy policy, string ResourceGroupNam
             }
         }
 
-        public BlobInventoryPolicy ParseBlobInventoryPolicy()
+
+        public Track2.BlobInventoryPolicyData ParseBlobInventoryPolicy()
         {
-            List<BlobInventoryPolicyRule> invRules = ParseBlobInventoryPolicyRules(this.Rules);
+            List<Track2Models.BlobInventoryPolicyRule> invRules = ParseBlobInventoryPolicyRules(this.Rules);
+
+            Track2Models.BlobInventoryPolicySchema policySchema = new Track2Models.BlobInventoryPolicySchema(
+                this.Enabled,
+                Track2Models.InventoryRuleType.Inventory,
+                invRules);
 
-            BlobInventoryPolicySchema policySchema = new BlobInventoryPolicySchema()
+            Track2.BlobInventoryPolicyData data = new Track2.BlobInventoryPolicyData
             {
-                Enabled = this.Enabled,
-                //Destination = this.Destination,
-                Rules = invRules
+                Policy = policySchema,
```

Comment:
It looks still still missing ID, Name, LastModified...

Reply:
Seems like we cannot assign those values, including id, name, and lastModifiedOn, as the constructor is internal. I'll record this issue. 

![image](https://user-images.githubusercontent.com/100746763/175449785-3ec60f39-35cd-443e-bb34-1fb5a31fcdfd.png)

Reply:
This might not be an issue, since this function is only used in set-AzstorageStorageBlobInventoryPolicy, to parse the input the policy (PSH object) to the policy (SDK object) to set.

And I can understand SDK not allow to set them, since they should only be read from server, but not set by customer.

In a word, if PSH code works with it, I believe this will not be an issue.